### PR TITLE
Add gallery of images to home page

### LINKED ISSUE #13580: MOOSE has no gallery
## Reason
MOOSE-curious website visitors should have access to a summary of MOOSE's scope and accomplishments, as well as a gallery showcasing results obtained with MOOSE.

## Design
This gallery should be high-visibility, probably on the front page.

## Impact
More MOOSE users.

### PR DESCRIPTION
**Pull request by aeslaughter**
(closes #13580)

### CODE REVIEW

**Comment by aeslaughter** [comment]
@cticenhour You can add your movie/image to this PR, the actual image should be in the large_media repo. I created a placeholder for you in the markdown.

**Comment by lindsayad** [comment]
Looks like I’ll have to get some laser welding in here!

File: modules/doc/content/index.md [hunk 1]
```diff
@@ -12,27 +12,82 @@
...
 !row-end!
 
+
+!gallery! large=6
+!card! gallery/corner_flow.gif title=Porous Flow around Corner style=height:320px
+Transient flow around corner using porous flow approximation of Euler equations with variable porosity,
+see [Pronghorn: Porous media thermal-hydraulics for reactor applications](https://escholarship.org/uc/item/61k9r05w).
+!card-end!
+
+!card! gallery/elder.gif title=Density driven, porous flow style=height:320px
```
Comment (friedmud) [inline modules/doc/content/index.md]:
```suggestion
!card! gallery/elder.gif title=Density driven, porous flow with adaptivity style=height:320px
```

File: modules/doc/content/index.md [hunk 2]
```diff
@@ -12,27 +12,82 @@
...
+Density driven, porous flow simulation of the Elder problem using [Falcon](https://github.com/idaholab/falcon) (*credit: Robert Podgorney (INL)*).
```
Comment (friedmud) [inline modules/doc/content/index.md]:
```suggestion
Density driven, porous flow simulation of the Elder problem using [Falcon](https://github.com/idaholab/falcon).  Mesh adaptivity is used to accurately capture the moving fronts. (*credit: Robert Podgorney (INL)*).
```

File: modules/doc/content/index.md [hunk 3]
```diff
@@ -12,27 +12,82 @@
...
+media of closely packed steel spheres, see [MultiApps/index.md].
+!card-end!
+
+!card! gallery/densification.gif title=3D Densification of snow style=height:300px
+Densification of a 3D snow pack using empirical, density based continuum model using [Pika](https://github.com/idaholab/pika).
+!card-end!
+!gallery-end!
+
+!gallery! large=6
+!card! gallery/snow.gif title=Dendritic Crystal Growth style=height:480px
```
Comment (friedmud) [inline modules/doc/content/index.md]:
```suggestion
!card! gallery/snow.gif title=Dendritic crystal growth with adaptivity style=height:480px
```

**Review by friedmud — APPROVED** [review]

**Pull request merged**